examples/mongocxx/query_projection.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/options/find.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/uri.hpp>
+
+using bsoncxx::builder::stream::document;
+using bsoncxx::builder::stream::open_document;
+using bsoncxx::builder::stream::close_document;
+using bsoncxx::builder::stream::finalize;
+
+int main(int, char **) {
+  mongocxx::instance inst{};
+  mongocxx::client conn{mongocxx::uri{}};
+
+  auto coll = conn["test"]["foo"];
```

Comment:
Maybe slightly better to use the name of the test binary as the collection, in case we want to run examples concurrently during test.